File: net/lib/rules/StringRules.cs
namespace Albert.Lib.Rules;


public class StringMinRule(int min) : IBaseRule<string>
{
    public string ErrorMessage() => $"must be at least {min} characters";

    public bool Validate(string? value) => value?.Length >= min;
}

public class StringMaxRule(int max) : IBaseRule<string>
{
    public string ErrorMessage() => $"must be no more than {max} characters";

    public bool Validate(string? value) => value?.Length <= max;
}

File: js/lib/rules/stringRules.ts
import { BaseRule } from '.';

export class StringMinRule implements BaseRule<string> {
  private _min: number;

  constructor(min: number) {
    this._min = min;
  }

  errorMessage() {
    return `must be at least ${this._min} characters`;
  }

  validate(value?: string) {
    return (value?.length || 0) >= this._min;
  }
}


export class StringMaxRule implements BaseRule<string> {
  private _max: number;
  constructor(max: number) {
    this._max = max;
  }

  errorMessage() {
    return `must be no more than ${this._max} characters`;
  }

  validate(value?: string) {
    return (value?.length || 0) <= this._max;
  }
}